### DISCUSSION TITLE
[v4] Self relationship manyThrough getters

### DISCUSSION THREAD

**Discussion by Neewd** [adonisjs/core#1884] [Help]
Hey community :) 

I have a question about data retrieval with self relationship. 
My need / my model is the following. 

I have a parent (people) who has children (people also), with a parent_id referencing the same table.
And these people can take picture,

So the people have a relationship to retrieve these pictures. 
But my need to is to retrieve all the pictures for him and also all his children. 

How can I achieve that (without raw querying ^^') ?

Thanks guys for  your help :)

**Comment by thetutlage** [comment]
You need to share some code of what you are tried so far

**Comment by Neewd** [comment]
Yeah sorry, it's because it was more in the idea process than already done but hurthing process :) 

For example I have this Country model 

```
class Country extends Model {
    
    countries() {
        return this.hasMany('App/Models/Country', 'parent_id', 'id')
    }

    posts() {
        return this.manyThrough('App/Models/Location', 'posts')
    }

}
```

In my controller, before the children / parent needs I was retrieving countries posts like so 
` const country = await Country.findBy('id', params.countryId)
await country.posts().isActive().paginate(page, limit)
`

But now, I would like to retrieve for a parent, all the country children posts as well. 
Should I do a country.posts().with('countries') ... ?

**Comment by Neewd** [comment]
Or maybe, I do this in the wrong way ?

**Comment by Neewd** [comment]
If I do this 
```
            const countries = await Country.query()
                        .where('id', '=', params.countryId)
                        .with('countries')
                        .with('countries.posts')
                        .with('countries.posts.location')
                        .with('posts.location')
                        .with('posts')
                        .paginate(page,limit)
```

I will get my countries and their posts, and also their children with own data. 

But what I would like to achieve (if possible) is to have a way to retrieve all the posts() from the direct relation and the children relation. (self)

Like a allPosts method in the model who will concat the direct post and the post from the self-relation.

Is it possible ?
